### DISCUSSION TITLE
FreeBsd installation with BastilleFile, exception with templates

### DISCUSSION THREAD

**Discussion by reyman** [photoprism/photoprism#2679] [Q&A]
Hi,

I'm trying to run photoprism on FreeBSD with Bastille Jail and [huo-ju](https://github.com/huo-ju/photoprism-freebsd-ports) ports.

My current Bastille File that compile libtensor and photoprism port is : 

```
PKG ffmpeg p5-Image-ExifTool bash bazel029 git gmake go npm wget

CMD portsnap fetch extract

CP usr /

# compiling libtensorflow
CMD mkdir -p /usr/ports/science/
CMD rm -rf /usr/ports/science/libtensorflow1
CMD cd  /usr/ports/science/ && git clone https://github.com/psa/libtensorflow1-freebsd-port libtensorflow1
#CMD cp /usr/local/etc/photoprism/extra-patch-third_party_repo.bzl /usr/ports/science/libtensorflow1/files/patch-extra-patch-third_party_repo.bzl

CMD make -C /usr/ports/science/libtensorflow1 package -DBATCH

CMD rm -rf photoprism-freebsd-port
CMD git clone https://github.com/huo-ju/photoprism-freebsd-port
CMD cd photoprism-freebsd-port && make config
CMD cd photoprism-freebsd-port && make -DBATCH
CMD cd photoprism-freebsd-port && make install

SYSRC photoprism_enable=YES

FSTAB /data/photoprism/ data/photoprism nullfs rw 0 0

CMD chown -R photoprism:photoprism /data/photoprism/

```
 
The actual ports made by @huo-ju specify some basic option for photoprism(assets, storage), but i want to customize more thing later, so i decide to use an `.yml` to override default.

I modify `/usr/local/etc/rc.d/photoprism` command_args that call photoprism by : 

`photoprism --defaults-yaml=/usr/local/etc/photoprism/options.yml --trace start`

My option.yml is 

```
HttpPort: 2342
HttpHost: 127.0.0.1
debug: true

AssetsPath: /data/photoprism/assets
ConfigPath: /usr/local/etc/photoprism
StoragePath: /data/photoprism/storage
OriginalsPath: /data/photoprism/storage/originals
ImportPath: /data/photoprism/storage/imports

AdminPassword: xxx
```

Running `service photoprism start` and looking into `/var/log/messages` i see that default value are overidded BUT i have this exception : 
```
level=error msg="html/template: no files named in call to ParseFiles"
```

That strange because my /data/photoprism folder is populated with files, including assets.

The photoprism version used by ports is `220730-0e1222c`

Any idea ?

**Comment by lastzero** [comment]
Probably didn't copy all the assets?

https://github.com/photoprism/photoprism/blob/d07e263a81b88ef5f1ccb38158aab2203da72bc1/Makefile#L93-L110

**Reply by reyman** [reply]
Thanks, i have only folder and any files in /assets/ so there is something missing. 

I push an issue here for discussion with author of the ports : https://github.com/huo-ju/photoprism-freebsd-port/issues/42

**Reply by reyman** [reply]
The assets builded by ports are localized into /var/db/photoprism/assets , i'm not sure this is normal : 

![Capture d’écran du 2022-09-04 15-41-05](https://user-images.githubusercontent.com/466329/188316635-728405b4-a78d-41aa-9532-9b28f8335d9a.png)

These files need to be into `/data/photoprism/assets` ?

**Reply by reyman** [reply]
I test by copying all this files into `/photoprism/assets` and everything run :+1: , so there is a little bug into the ports..
